gpkit/keydict.py

```diff
@@ -121,9 +123,8 @@ def __setitem__(self, key, value):
         key, idx = self.parse_and_index(key)
         if key not in self.keymap:
             self.keymap[key].add(key)
-            if hasattr(key, "keys") and self.keymapping:
-                for mapkey in key.keys:
-                    self.keymap[mapkey].add(key)
+            if hasattr(key, "keys"):
+                self._unmapped_keys.add(key)
```

Comment:
1) `update` is overloaded to use `__setitem_`, dictionaries have no `add`
2) the laziness occurs on `parse_and_index`, not `__setitem__`: this line just adds to a queue of "hey these should maybe be mapped at some point"